File: scripts/get_char_count.js
const pdfreader = require("pdfreader");
const childProcess = require("child_process")
const path = require("path")
const pdflatex = require(path.resolve(__dirname, "../.vscode/settings.json"))["latex-workshop.latex.tools"].filter(x=>x.name === "pdflatex")[0] || null
pdflatex.args = pdflatex.args.map(x=>{
    if(x === "%DOC%"){
        return "small_master.tex"
    }
    return x;
})

let i = -1;

new pdfreader.PdfReader().parseFileItems(path.resolve(__dirname, "../master.pdf"), (err, item)=>{
    if(!item){
        console.log(i);
        return;
    }
    if(!item.text ){
        return;
    }    
    i += item.text.length +1;

})

File: scripts/get_char_count.rb
require "json"
require "pdf-reader"

if ARGV.include?("--fast")
    Process.detach(Process.spawn("ruby " + __FILE__))
else
    latex_file = "small_master"

    command = JSON.parse(File.read(".vscode/settings.json"))["latex-workshop.latex.tools"].find {|x| x["name"] == "pdflatex"}
    args = command["args"]
    command = command["command"]
    args = args.map {|x| x == "%DOC%" ? latex_file + ".tex" : x}
    `#{command} #{args.join(" ")}`

    pdf = PDF::Reader.new(latex_file + ".pdf")
    char_count = pdf.pages.map {|x| x.text}.join(" ").gsub(/\s+/, " ").length
    File.write("current_char_count.tmp", char_count)
    File.write("current_page_count.tmp", (char_count / 2400).round)
end
exit
